### ISSUE TITLE
Cannot exit Customizer by single clicking the X at top left

### ISSUE THREAD

**Issue by designsimply** [Automattic/wp-desktop#932]
Steps to reproduce:

1. Go to My Site > My Home > Quick Links > Edit menus.
1. Single click on the X at the top left corner.
1. Note the customizer does not close.
1. Double click on the X at the top left corner.

**Result:** you must double click the X at top left in order to leave the Customizer but I expected that a single click  would work. ([3m17s](https://cloudup.com/cbh1YbKO0_B), [54s](https://cloudup.com/c0JptZqU5t0)) 

<img src="https://user-images.githubusercontent.com/1119271/85634233-28f83380-b638-11ea-99db-bde9537442e7.png" width="33%" alt="edit-menus"> <img width="33%" alt="Screen Shot 2020-06-24 at Jun 24 4 34 35 PM" src="https://user-images.githubusercontent.com/1119271/85634479-b5a2f180-b638-11ea-8c9d-8dd0eacc02f9.png"> <img src="https://user-images.githubusercontent.com/1119271/85634238-2c8bba80-b638-11ea-9b76-caa999dd430c.png" width="33%" alt="sorry-the-customizing-tools-did-not-load-correctly">

<p style="clear:both;"><sup>Tested with WPDesktop 6.0.0-beta3 on macOS 10.15.4.</sup></p>

**Comment by designsimply** [comment]
Well, I just deleted and reinstalled 6.0.0-beta3 and now I'm not seeing the problem happen any more. If no one else can reproduce this one then let's close it.

**Comment by nsakaimbo** [comment]
> Well, I just deleted and reinstalled 6.0.0-beta3 and now I'm not seeing the problem happen any more.

Yeah that's odd. Let's keep an eye on this one. At the very least, it's non-blocking (i.e. a double-click resolves it, and it doesn't put the app in a broken state).

**Comment by rachelmcr** [comment]
I ran into this with the same steps after creating a new account and site in 6.0.1-beta2. I tried repeatedly to click the X to close the customizer and nothing happened. I didn't realize that double-clicking would work, so I close the app and relaunched it.

After relaunching the app, it tried and failed to reopen the customizer (I got the error that the customizing tools didn't load correctly). After that, the first few times I opened the customizer I had to click twice (two single clicks, not a double click) to get the customizer to close, but that worked. After doing that a few times, the customizer started to close immediately the first time I clicked the X.

**Comment by designsimply** [comment]
This came up again in WPDesktop 6.1.0-beta3 testing on Windows 10 Pro. 

> On a non-Automattic account it seemed, that clicking on Edit menus brought me to the customizer as it suppose to, but I needed to click the x on the left top corner to get back to My Home twice, which did not happen on my Automattic account.

<sup>(internal reference: p5T066-1wN-p2 hat tip Sander)</sup>

**Comment by sarahfg** [comment]
Also found this issue repeating on WPDesktop 6.2.0 testing on Mac OS 10.15.7 

![](https://d.pr/i/XWYcMW+)

(Image link: https://d.pr/i/XWYcMW )

**Comment by nsakaimbo** [comment]
I've spent some time trying to reproduce this one (`v6.2.0-beta1`), but haven't had much luck so far.

- Based on @designsimply's video, this bug appears to occur when a `customize.php?...` request is canceled. Another possible clue (not sure if a red herring) is the customizer seems to load _very_ slowly when this happens.
- I tested the "Quick Links > Edit Menus" and "Quick Links > Customize them" flows for different types of sites/accounts (Automattic, Simple and Atomic) and wasn't able to reproduce.
- Thinking that perhaps network flakiness might make this state more likely, I also simulated poor network conditions on MacOS with the [Network Link Conditioner](https://nshipster.com/network-link-conditioner/) ("Very Bad Network" setting) to try to force some requests to fail. Still no luck - although things loaded slowly, the "Close" button continued to work as expected.

Perhaps this occurs for a specific account configuration (although unsure what that might be, at the moment). At least it doesn't seem to be happening consistently for everyone across the board, however we've had a number of these reports so it's certainly happening more than infrequently.

For future reference, when you encounter this again you should be able to use the keyboard shortcut `CMD + 1` (`CTRL + 1` on Windows), or use the application's `View` menu, to navigate to the Calypso's main landing page. (Obviously this still isn't ideal for end-users, but is useful to know during manual testing.) cc: @sarahfg

**Comment by sarahfg** [comment]
@nsakaimbo I'm not sure if this helps, but in my testing I had to click a few links first before I could get the behavior to happen. I went to My Home> Quick Links then opened each and came back to Quick Links, from first to last to then I get “stuck” on the customizer. Of course now I can't reproduce it either any more!

**Comment by designsimply** [comment]
I found I can reproduce this pretty consistently in 6.5.0-beta1 using these steps:

1. Open the WordPress.com Desktop App and go to My Site > My Home.
1. Close the app.
1. Re-open the app.
1. Go to My Site > My Home, find the Quick Links section and tap "Edit Menus."
1. Wait for the Customizer to load.
1. Tap the "x" at top left.

**Result:** cannot exit the Customizer if I access it using a deep link in a new session. ([3m47s](https://cloudup.com/coiWVPU6xKW))

![customizer-exit-button-highlighted](https://user-images.githubusercontent.com/1119271/100486966-adf1e800-30c3-11eb-918e-0c45c9c4ad9d.png)
<sup>Tested with WPDesktop 6.5.0-beta1 on macOS 10.15.6.</sup>

**Note:** I can no longer double tap the x to get out like it mentions in step 4 on the original issue, but I can also only trigger this bug if I try a deep link to something in the Customizer in a new session of the app where I have not previously opened the Customizer in the same session so I think that means the priority should stay normal or even go to low.

**Comment by nsakaimbo** [comment]
Thanks for the update, @designsimply. I will go ahead and transfer this issue to the wp-calypso repo, so we can keep track of it there.
